simapp/app.go

```diff
@@ -450,7 +451,9 @@ func (app *SimApp) EndBlocker(ctx sdk.Context, req abci.RequestEndBlock) abci.Re
 // InitChainer application update at chain initialization
 func (app *SimApp) InitChainer(ctx sdk.Context, req abci.RequestInitChain) abci.ResponseInitChain {
 	var genesisState GenesisState
-	app.cdc.MustUnmarshalJSON(req.AppStateBytes, &genesisState)
+	if err := tmjson.Unmarshal(req.AppStateBytes, &genesisState); err != nil {
+		panic(err)
```

Comment:
Ouch, can we return an error and handle it somewhere higher, please?

Reply:
This _must_ panic (it used to panic too). An error that happens in `InitChain()` is unrecoverable, as such it must cause the program to crash and make as much noise as possible.